src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
```

Comment:
This part unclear and has low readability
Also don't be afraid to use additional variables where they allow to make the code more readable, for example:
```suggestion
children.forEach(child => {
  const childMother = findMother(child, people);
  const bornDifference = child.born - childMother.born;

  sum += bornDifference;
};
```

Try to rewrite it with reduce